src/server/db_slice.cc

```diff
@@ -1160,7 +1160,6 @@ uint64_t DbSlice::RegisterOnChange(ChangeCallback cb) {
 
 void DbSlice::FlushChangeToEarlierCallbacks(DbIndex db_ind, Iterator it, uint64_t upper_bound) {
   FetchedItemsRestorer fetched_restorer(&fetched_items_);
-  std::unique_lock<LocalBlockingCounter> lk(block_counter_);
```

Comment:
Don't we need `std::unique_lock lk(local_mu_);` here?

Reply:
nope, it's called in two places:

1. Within the callback passed to Traverse which locks this mutex
2. Within the `BucketSaveCb` which again is passed to Traverse

Reply:
ok.
Just FYI, the compiler can assist with these things - add `ABSL_LOCKS_EXCLUDED` to functions that _will_ lock, and `ABSL_LOCK_REQUIRED` for locks that assume that a lock is already locked (perhaps I misspelled some there)
This prevents errors, although I do not think currently there is one.

Reply:
I just saw the comment otherwise I have added them. Will include them once I revert the change with the flag being set to 1